src/qt/sendcoinsdialog.cpp

```diff
@@ -642,16 +642,14 @@ void SendCoinsDialog::updateSmartFeeLabel()
     updateCoinControlState(coin_control);
     coin_control.m_feerate = boost::none; // Explicitly use only fee estimation rate for smart fee labels
     FeeCalculation feeCalc;
-    bool conservative_estimate = CalculateEstimateType(FeeEstimateMode::UNSET, coin_control.signalRbf);
-    CFeeRate feeRate = ::feeEstimator.estimateSmartFee(*coin_control.m_confirm_target, &feeCalc, ::mempool, conservative_estimate);
+    CFeeRate feeRate = CFeeRate(CWallet::GetMinimumFee(1000, coin_control, ::mempool, ::feeEstimator, &feeCalc));
 
     // show the estimated required time for confirmation
     ui->confirmationTargetLabel->setText(GUIUtil::formatDurationStr(*coin_control.m_confirm_target * Params().GetConsensus().nPowTargetSpacing) + " / " + tr("%n block(s)", "", *coin_control.m_confirm_target));
 
-    if (feeRate <= CFeeRate(0)) // not enough data => minfee
-    {
-        ui->labelSmartFee->setText(BitcoinUnits::formatWithUnit(model->getOptionsModel()->getDisplayUnit(),
-                                                                std::max(CWallet::fallbackFee.GetFeePerK(), CWallet::GetRequiredFee(1000))) + "/kB");
+    ui->labelSmartFee->setText(BitcoinUnits::formatWithUnit(model->getOptionsModel()->getDisplayUnit(), feeRate.GetFeePerK()) + "/kB");
+
+    if (feeCalc.reason == FeeReason::FALLBACK) {
```

Comment:
I think the behavior in rare edge cases is maybe not precisely correct, but there is likely no harm in it.  The only thing you are missing is a warning that fee estimation is not up to date yet, and before this PR that could already happen if there was a mempool min fee (in which case you wouldn't even get maxed with the fallback fee).  After this PR, you only miss the warning if your fallback fee is less than mempool min fee or minRelayTxFee or minTxFee.  The first case is a strict improvement over prior behavior. The second two cases can only result from very poor configuration of command line options.  In short, I think it could be improved to report even more precise information to the user but it's outside the scope of what I wanted to do here.

I can expand commit description, but the idea is the displays should correspond to the fees that your actually about to put on the transaction if you click send.  I think it's essentially a bug that it didn't do that before.